apps/gdalenhance.cpp

```diff
@@ -485,24 +496,27 @@ static CPLErr EnhancerCallback(void *hCBData, int nXOff, int nYOff, int nXSize,
         return eErr;
     }
 
-    int nPixelCount = nXSize * nYSize;
+    const size_t nPixelCount = static_cast<size_t>(nXSize) * nYSize;
     int bHaveNoData;
-    float fNoData =
+    const float fNoData =
         static_cast<float>(psEInfo->poSrcBand->GetNoDataValue(&bHaveNoData));
-    double dfScale =
+    const double dfScale =
         psEInfo->nLUTBins / (psEInfo->dfScaleMax - psEInfo->dfScaleMin);
 
-    for (int iPixel = 0; iPixel < nPixelCount; iPixel++)
+    for (size_t iPixel = 0; iPixel < nPixelCount; iPixel++)
     {
         if (bHaveNoData && pafSrcImage[iPixel] == fNoData)
         {
             pabyOutImage[iPixel] = static_cast<GByte>(fNoData);
             continue;
         }
 
-        int iBin = static_cast<int>(
-            (pafSrcImage[iPixel] - psEInfo->dfScaleMin) * dfScale);
-        iBin = std::max(0, std::min(psEInfo->nLUTBins - 1, iBin));
+        const double dfBin =
+            (pafSrcImage[iPixel] - psEInfo->dfScaleMin) * dfScale;
+        const int iBin = (!(dfBin > 0)) ? 0
```

Comment:
Split into multiple statements for legibility?

Reply:
done